packages/pds/src/migrate-script/index.ts

```diff
@@ -0,0 +1,417 @@
+) => {
+  console.log(`reserve (${status.did}, ${status.phase})`)
+  if (status.phase < TransferPhase.reservedKey) {
+    const signingKey = await reserveSigningKey(pds, status.did)
+    status.signingKey = signingKey
+    status.phase = TransferPhase.reservedKey
+    await updateStatus(db, status)
+  }
+
+  console.log(`import (${status.did}, ${status.phase})`)
+  if (status.phase < TransferPhase.initImport) {
+    const importedRev = await doImport(ctx, db, pds, status.did)
+    if (importedRev) {
+      status.importedRev = importedRev
+    }
+    status.phase = TransferPhase.initImport
+    await updateStatus(db, status)
+  }
+
+  console.log(`transfer (${status.did}, ${status.phase})`)
+  if (status.phase < TransferPhase.transferred) {
+    const importedRev = await lockAndTransfer(ctx, db, pds, status)
+    status.importedRev = importedRev
+    status.phase = TransferPhase.transferred
+    await updateStatus(db, status)
+  }
+
+  console.log(`prefs (${status.did}, ${status.phase})`)
+  if (status.phase < TransferPhase.preferences) {
+    try {
+      await transferPreferences(ctx, pds, status.did)
+    } catch (err) {
+      await db
+        .insertInto('failed_pref')
+        .values({ did: status.did })
+        .onConflict((oc) => oc.doNothing())
+        .execute()
+    } finally {
+      status.phase = TransferPhase.preferences
+      await updateStatus(db, status)
+    }
+  }
+
+  console.log(`takedowns (${status.did}, ${status.phase})`)
+  if (status.phase < TransferPhase.takedowns) {
+    await transferTakedowns(ctx, db, pds, status.did, adminToken)
+    status.phase = TransferPhase.completed
+    await updateStatus(db, status)
+  }
+}
+
+const updateStatus = async (db: MigrateDb, status: Status) => {
+  return db
+    .updateTable('status')
+    .set({ ...status })
+    .where('did', '=', status.did)
+    .execute()
+}
+
+const reserveSigningKey = async (
+  pds: PdsInfo,
+  did: string,
+): Promise<string> => {
+  const signingKeyRes =
+    await pds.agent.api.com.atproto.server.reserveSigningKey({ did })
+  return signingKeyRes.data.signingKey
+}
+
+const doImport = async (
+  ctx: AppContext,
+  db: MigrateDb,
+  pds: PdsInfo,
+  did: string,
+  since?: string,
+) => {
+  const storage = new SqlRepoStorage(ctx.db, did)
+  const root = await storage.getRootDetailed()
+  if (!root) {
+    throw new Error(`repo not found: ${did}`)
+  }
+  if (since && root.rev === since) {
+    return
+  }
+  const carStream = await storage.getCarStream(since)
+
+  const importRes = await axios.post(
+    `${pds.url}/xrpc/com.atproto.temp.importRepo`,
+    carStream,
+    {
+      params: { did },
+      headers: { 'content-type': 'application/vnd.ipld.car' },
+      decompress: true,
+      responseType: 'stream',
+    },
+  )
+
+  for await (const log of importRes.data) {
+    if (typeof log === 'string') {
+      const lines = log.split('\n')
+      for (const line of lines) {
+        if (line.indexOf('failed to import blob') > 0) {
```

Comment:
If this string is at the beginning of the line (i.e. at index `0`) we'd actually miss it!  I like to use `includes()` to avoid the footgun:
```suggestion
        if (line.includes('failed to import blob')) {
```

Reply:
oh duh good catch